optimized load_image_sub_file() [LLM assisted]

mentos987 opened the pull request:
This introduces a new cache system for **intermediate modified surfaces** within `load_image_sub_file()`. This significantly reduces duplicate work by avoiding the repeated, expensive process of cloning a base surface and applying a complex modification stack (IPFs like colour change, opacity shift, scale, ..) when drawing multiple tiles from the same modified, uncut source image.

It also avoids calling some heavy functions when not needed.

The changes mostly focus on modified multihex images that gets cut up into hex sizes, such as water tiles.

### Performance Gains

I have tested this by comparing time spent in `load_image_sub_file()` before and after this code change. I ran one medium load (a bunch of sea tiles) and one heavy (large custom multihex images).

The report shows how much time was spent in the function in the last 5 seconds of scrolling over the map on zoom level 3.

Before code change:
```
     Medium load:
          [PERF REPORT] Load Time (Last 5s) = 576.626 ms
     Heavy load:
          [PERF REPORT] Load Time (Last 5s) = 3163.435 ms
```
          
After code change:
```
     Medium load:
          [PERF REPORT] Load Time (Last 5s) = 428.011 ms
     Heavy load:
          [PERF REPORT] Load Time (Last 5s) = 720.124 ms
```
The results varied a lot (+-20%) but the average was consistent with what you see above.

**Importantly, the greatest time savings occur during heavy loading, which corresponds to the most noticeable lag spikes.**

The extra cache is limited to 200 MB but even in my heavy test it didn't go above 16 MB. If you place 1 huge image that covers a 200x200 map you may get above that limit.

soliton- commented on src/picture.cpp:
```diff
@@ -205,6 +205,100 @@ parsed_data_URI::parsed_data_URI(std::string_view data_URI)
+}
+
+/**
+ * Prunes the modified surface cache based on LRU policies (cache over 200MB).
```
I'm talking about the code comment repeating the currently set limit. IMO that is simply asking to become wrong if the limit is changed.

mentos987 commented:
Vultraz has requested that this wait until after SDL3 merge, otherwise it is ready to merge.

mentos987 committed:
Optimize load_image_sub_file() with an intermediate surface cache

Large images cut into several hex tiles (e.g. water) previously had
their modifications re-applied once per tile. Cache the modified but
uncut surface in a size-bounded TTL+LRU cache so repeated cutouts of
the same source reuse it instead of redoing the work.

mentos987 commented:
@Pentarctagon @Vultraz  We should try to add this before 1.20 . It achieves major lag reductions.
